gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MysqlMultiActiveLeaseArbiter.java

```diff
@@ -104,32 +116,50 @@ protected interface CheckedFunction<T, R> {
       + "VALUES(1, ?, ?) ON DUPLICATE KEY UPDATE epsilon=VALUES(epsilon), linger=VALUES(linger)";
   protected static final String WHERE_CLAUSE_TO_MATCH_KEY = "WHERE flow_group=? AND flow_name=? AND flow_action=?";
   protected static final String WHERE_CLAUSE_TO_MATCH_ROW = WHERE_CLAUSE_TO_MATCH_KEY
-      + " AND event_timestamp=? AND lease_acquisition_timestamp=?";
-  protected static final String SELECT_AFTER_INSERT_STATEMENT = "SELECT event_timestamp, lease_acquisition_timestamp, "
-    + "linger FROM %s, %s " + WHERE_CLAUSE_TO_MATCH_KEY;
+      + " AND event_timestamp=CONVERT_TZ(?, '+00:00', @@session.time_zone)"
+      + " AND lease_acquisition_timestamp=CONVERT_TZ(?, '+00:00', @@session.time_zone)";
```

Comment:
don't we prefer to store and retrieve always as UTC?  if so, is the issue that `DEFAULT CURRENT_TIMESTAMP(3)` is what first sets us on to using the session TZ?  alternatively could we convert it to be UTC when it's initialized that way?  if we could, there wouldn't be a need for conversion here, would there be?

e.g. do we want everywhere want `UTC_TIMESTAMP(3)`, rather than `CURRENT_TIMESTAMP(3)`?

Reply:
From the documentation, the timestamp stored is in UTC but upon retrieval/or interpreting timestamps provided it automatically converts it to the current time zone (determined by the server's time). On our side we are utilizing all UTC timestamps so I convert it to session timezone as MySQL will interpret it as session timezone. I *think* using `UTC_TIMESTAMP` as the default won't change that so internally I am using session time zone with `CURRENT_TIMESTAMP` . I tried using `UTC_TIMESTAMP(3)` before and got a syntax error below. Debugging the timezone issues on the unit tests is what took the majority of time in this change and this configuration with the unit tests is what I could get to work and seem like it's using the same timezone everywhere. 

`Caused by: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'UTC_TIMESTAMP(3), lease_acquisition_timestamp TIMESTAMP(3) NULL DEFAULT UTC_TIME' at line 1`

Reply:
aren't we on mysql 8?  if so, how about this - https://dba.stackexchange.com/a/279140 ?
```
event_timestamp TIMESTAMP(3) NOT NULL DEFAULT (UTC_TIMESTAMP)
```
?

Reply:
I tried variations of this and got either syntax errors or `Invalid default value for 'event_timestamp'. I'll instead set the session timezone to UTC before reading or inserting. 

```
event_timestamp TIMESTAMP(3) NOT NULL DEFAULT UTC_TIMESTAMP
event_timestamp TIMESTAMP(3) NOT NULL DEFAULT (UTC_TIMESTAMP)
event_timestamp TIMESTAMP(3) NOT NULL DEFAULT (UTC_TIMESTAMP(3))
event_timestamp TIMESTAMP(3) NOT NULL DEFAULT UTC_TIMESTAMP(3)
```

Reply:
Set default to `NULL` and use `CURRENT_TIMESTAMP(3)` upon insertion or updates bc the timestamps are automatically interpreted in session timezone. 

Including for documentation purposes that I also tried setting the session timezone before making insertions/updates/table creation but had trouble executing _multi-query_ statements with a `SET` and `SELECT` because the latter needs to be executed with `preparedStatement.executeQuery` to return a `ResultSet` but multiple queries are only allowed with `preparedStatement.execute`. The JDBC connection url also needs to also `multi-query statements` as the default connection url does not allow it. The use of the former with the multi-statement query results in a SQL error 

`java.sql.SQLException: ResultSet is from UPDATE. No Data.`